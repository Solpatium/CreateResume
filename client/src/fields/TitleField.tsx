import {Field, IFieldProps, IFieldState} from './Field';
import * as React from 'react';
import { Input, Form } from 'antd';
const FormItem = Form.Item;

interface ITextFieldProps extends IFieldProps {
    value?: string
}

interface ITextFieldState extends IFieldState {
    value: string
}

export default class TitleField extends Field<ITextFieldProps, ITextFieldState> {
    constructor(props: ITextFieldProps) {
        super(props)
        this.state = {
            hidden: false,
            id: `field-${Field.counter}`,
            value: props.value ? props.value : ''
        }
    }

    public get name() { return 'Section title' };

    public get content(): React.ReactNode {
        return (
            <FormItem label="Title">
                <Input placeholder="Education" value={this.state.value} onChange={this.update} />
            </FormItem>
        )
    }

    public update = (event: React.ChangeEvent<HTMLInputElement>) => {
        this.setState({value: event.target.value});
        this.notifyChange()
    }
}