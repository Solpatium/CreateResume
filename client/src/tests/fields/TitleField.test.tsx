import React from 'react';
import TitleField from '../../fields/TitleField';
import renderer from 'react-test-renderer';
import { DragDropContext } from 'react-beautiful-dnd';

test('WorkField renders properly', () => {
    const component = renderer.create(
        <DragDropContext onDragEnd={()=>null}><TitleField index={0}/></DragDropContext>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});