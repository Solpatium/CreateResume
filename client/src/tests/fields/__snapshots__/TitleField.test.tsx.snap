// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkField renders properly 1`] = `
<div
  aria-roledescription="Draggable item. Press space bar to lift"
  className="field"
  data-react-beautiful-dnd-drag-handle="0"
  data-react-beautiful-dnd-draggable="0"
  draggable={false}
  onBlur={[Function]}
  onDragStart={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  onTouchStart={[Function]}
  style={
    Object {
      "transform": null,
      "transition": null,
    }
  }
  tabIndex={0}
>
  <div
    className="ant-card ant-card-bordered"
  >
    <div
      className="ant-card-head"
    >
      <div
        className="ant-card-head-wrapper"
      >
        <div
          className="ant-card-head-title"
        >
          Section title
        </div>
        <div
          className="ant-card-extra"
        >
          <div
            className="card-actions"
          >
            <button
              className="ant-btn ant-btn-danger ant-btn-circle ant-btn-icon-only"
              onClick={[Function]}
              type="button"
            >
              <i
                className="anticon anticon-close"
              />
            </button>
            <button
              className="ant-btn ant-btn-circle ant-btn-icon-only"
              onClick={[Function]}
              type="button"
            >
              <i
                className="anticon anticon-down"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-card-body"
      style={Object {}}
    >
      <div
        className="ant-row ant-form-item"
        style={undefined}
      >
        <div
          className="ant-form-item-label"
        >
          <label
            className=""
            htmlFor={null}
            onClick={[Function]}
            title="Title"
          >
            Title
          </label>
        </div>
        <div
          className="ant-form-item-control-wrapper"
        >
          <div
            className="ant-form-item-control"
          >
            <span
              className="ant-form-item-children"
            >
              <input
                className="ant-input"
                disabled={false}
                onChange={[Function]}
                onKeyDown={[Function]}
                placeholder="Education"
                type="text"
                value=""
              />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
